code/HabitTracker/app/src/main/java/com/example/habittracker/activities/eventlist/EventListActivity.java

```diff
@@ -69,41 +91,177 @@ public void onClick(View v) {
             /**
              * Override the onItemClick method. Once an item on the list is clicked,
              * we pass the selected event object to the floating fragment.
-             * @param adapter
-             * @param v
-             * @param position
-             * @param arg3
+             * @param adapter       {@code AdapterView<?>} the habit event adapter
+             * @param v             {@code View} view
+             * @param position      {@code int} index of selected item
+             * @param arg3          {@code long} arg3
              */
             public void onItemClick(AdapterView<?> adapter, View v, int position,
                                     long arg3) {
-                new AddEventFragment().newInstance().show(getSupportFragmentManager(),"EDIT_EVENT");
+                HabitEvent editE = (HabitEvent) adapter.getItemAtPosition(position);
+                new AddEventFragment().newInstance(editE).show(getSupportFragmentManager(),"EDIT_EVENT");
             }
         });
 
+        // set up snapshot listener
+        String usersColName = "Users";
+        String habitsColName = "Habits";
+        String habitEventsColName = "HabitEvents";
+        String DB_TAG = "DatabaseManager";
+
+        ArrayList<String> habit_list = new ArrayList<>();
+
+        DatabaseManager dm = DatabaseManager.get();
+        CollectionReference colRef;
+        dm.getAllHabits("John_test_user", new HabitListCallback() {
```

Comment:
"Replace this line when share info is up"

Reply:
Fixed